backend/addresses/serializers.py

```diff
@@ -0,0 +1,81 @@
+import re
+from rest_framework import serializers
+from .models import Address
+from accounts.models import CustomerProfile
+
+EGYPT_GOVERNORATES = [
+    "Cairo", "Alexandria", "Giza", "Aswan", "Luxor", "Suez", "Port Said", "Ismailia", "Dakahlia",
+    "Qalyubia", "Sharqia", "Beheira", "Gharbia", "Monufia", "Faiyum", "Beni Suef", "Minya", "Sohag",
+    "Qena", "Assiut", "Red Sea", "New Valley", "North Sinai", "South Sinai", "Matruh", "Damietta",
+    "Kafr El Sheikh"
+]
+
+class AddressSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Address
+        fields = '__all__'
+        read_only_fields = ['user']
+
+    def validate_postal_code(self, value):
+        if not re.match(r"^\d{5}$", value):
+            raise serializers.ValidationError("Postal code must be 5 digits.")
+        return value
+
+    def validate_city(self, value):
+        if not value.strip():
+            raise serializers.ValidationError("City is required.")
+        return value
+
+    def validate_governorate(self, value):
+        if value not in EGYPT_GOVERNORATES:
+            raise serializers.ValidationError("Invalid governorate.")
+        return value
+
+    def validate_country(self, value):
+        if value.strip().lower() != "egypt":
+            raise serializers.ValidationError("Country must be 'Egypt'.")
+        return value
+
+    def create(self, validated_data):
+        user = self.context['request'].user
+        is_default = validated_data.pop("is_default", False)
+
+        address = Address.objects.create(user=user, **validated_data)
+
+        if is_default:
+            address.is_default = True
+            address.save(update_fields=["is_default"])
+
+            try:
+                profile = user.customerprofile
+                profile.default_shipping_address = address
+                profile.save(update_fields=["default_shipping_address"])
+            except CustomerProfile.DoesNotExist:
+                CustomerProfile.objects.create(user=user, default_shipping_address=address)
```

Comment:
Just a small update to meet the following business logic:
  - If the user has a `customer` role, they could add the address as a default address and update their
    `default_shipping_address` field.
  - The `create()` method should only be concerned with `creating an address`.
     - Creating a `CustomerProfile` is a user _onboarding concerns_, not an address-create concern.
    
Here my suggestion for the change:
```suggestion
            if user.role == "customer":
              try:
                  profile = user.customerprofile
                  profile.default_shipping_address = address
                  profile.save(update_fields=["default_shipping_address"])
              except CustomerProfile.DoesNotExist:
                  raise ValidationError("Customer profile must be created before assigning a default address.")
```

This ensures:
  - [x] That the address created as a default address only for users having a `customer` role.
  - [x] Doesn't blindly create a `CustomerProfile`, regardless of whether the `user.role` is `customer` or  not.
  - [x] Protects the business logic rules if validations are skipped for any reason.